solana-programs/staking-bridge/programs/staking-bridge/src/lib.rs

```diff
@@ -1,15 +1,271 @@
+     */
+    pub fn raydium_swap(
+        ctx: Context<RaydiumSwap>,
+        amount_in: u64,
+        minimum_amount_out: u64,
+        vault_nonce: u64,
+        staking_bridge_pda_bump: u8
+    ) -> Result<()> {
+        let accounts = ctx.accounts;
+        check_raydium_programs(accounts)?;
+        check_raydium_token_accounts(accounts)?;
+        check_raydium_pdas(
+            accounts,
+            vault_nonce
+        )?;
+        swap(
+            accounts,
+            amount_in,
+            minimum_amount_out,
+            staking_bridge_pda_bump
+        )?;
+        Ok(())
+    }
+
+    pub fn post_wormhole_message(
+        ctx: Context<PostWormholeMessage>,
+        nonce: u32,
+        amount: u64, // TODO: if amount greater than max amount of our PDA, do we send max or do we fail?
+        fee: u64, // TODO: do we need this? what if someone passes in a huge fee?
+        config_bump: u8,
+        wrapped_mint_bump: u8,
+        wrapped_meta_bump: u8,
+        authority_signer_bump: u8,
+        bridge_config_bump: u8,
+        emitter_bump: u8,
+        sequence_bump: u8,
+        fee_collector_bump: u8,
+        staking_bridge_pda_bump: u8,
+    ) -> Result<()> {
+        let accounts = ctx.accounts;
+        check_wormhole_programs(accounts)?;
+        check_wormhole_token_accounts(accounts)?;
+        check_wormhole_pdas(
+            accounts,
+            config_bump,
+            wrapped_mint_bump,
+            wrapped_meta_bump,
+            authority_signer_bump,
+            bridge_config_bump,
+            emitter_bump,
+            sequence_bump,
+            fee_collector_bump,
+        )?;
+        approve_wormhole_transfer(
+            accounts,
+            amount,
+            staking_bridge_pda_bump
+        )?;
+        execute_wormhole_transfer(
+            accounts,
+            nonce,
+            amount,
+            fee,
+            staking_bridge_pda_bump
+        )?;
         Ok(())
     }
 }
 
 #[derive(Accounts)]
-pub struct Initialize {}
+pub struct CreatePda<'info> {
+    #[account(
+        init,
+        seeds = [b"staking_bridge".as_ref()],
+        payer = payer,
+        bump,
+        space = 8
+    )]
+    /// CHECK: This is the PDA owned by this program. This account holds both SOL USDC and SOL AUDIO. It is used to swap between the two tokens. This PDA is also used to transfer SOL AUDIO to ETH AUDIO via the wormhole.
+    pub staking_bridge_pda: AccountInfo<'info>,
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(AnchorDeserialize, AnchorSerialize, Default)]
+pub struct Amounts {
+    pub amount_in: u64,
+    pub minimum_amount_out: u64,
+}
+
+#[derive(Accounts)]
+#[instruction(
+  _amount_in: u64,
+  _minimum_amount_out: u64,
+  _vault_nonce: u64,
+  staking_bridge_pda_bump: u8
+)]
+pub struct RaydiumSwap<'info> {
+    /// CHECK: This is the Raydium Liquidity Pool V4 program id
```

Comment:
what does `CHECK` mean here? for my own education. is this common anchor annotation?

Reply:
I had a similar question! https://www.anchor-lang.com/docs/the-accounts-struct See "Safety checks"

Also my comment: https://github.com/AudiusProject/audius-protocol/pull/5804#discussion_r1297839155